streams/src/test/java/org/apache/kafka/streams/tests/EosTestDriver.java

```diff
@@ -559,47 +601,36 @@ private static void verifyAllTransactionFinished(final KafkaConsumer<byte[], byt
                         exception.printStackTrace(System.err);
                         System.err.flush();
                         Exit.exit(1);
+                    } else {
+                        endMarkerOffset.put(new TopicPartition(metadata.topic(), metadata.partition()), metadata.offset());
+                        System.out.println("Appended verification record to topic-partition " + metadata.topic() + "-" + metadata.partition() + " at offset " + metadata.offset());
+                        System.out.flush();
                     }
                 });
             }
         }
 
-        final StringDeserializer stringDeserializer = new StringDeserializer();
```

Comment:
Instead of looking for the end-marker message per content (ie, comparing key and value), we now use the offset (that we now know) to see if we can get the `endOffset()` as expected in "read_committed" mode.